realm/realm-library/src/main/cpp/io_realm_RealmFileUserStore.cpp

```diff
@@ -31,11 +31,15 @@ Java_io_realm_RealmFileUserStore_nativeGetCurrentUser (JNIEnv *env, jclass)
 {
     TR_ENTER()
     try {
-
-        const std::shared_ptr<SyncUser> &user = SyncManager::shared().get_current_user();
-        if (user) {
-            return to_jstring(env, user->refresh_token().data());
-        } else {
+        try {
+            const std::shared_ptr<SyncUser> &user = SyncManager::shared().get_current_user();
+            if (user) {
+                return to_jstring(env, user->refresh_token().data());
+            } else {
+                return nullptr;
+            }
+        } catch (std::logic_error e) {
+            ThrowException(env, IllegalState, "Multiple users have been logged in. `currentUser` is only valid if one single user is logged in.");
```

Comment:
The OS has sent a quite meaningful exception message:
```c++
if (std::find_if(std::next(it), m_users.end(), is_active_user) != m_users.end()) {
        throw std::logic_error("Current user is not valid if more that one valid, logged-in user exists.");
```

Do we really need to rephrase it? Or the exception message in the OS should be changed directly.